lib/ui/src/modules/ui/components/layout/index.js

```diff
@@ -156,20 +157,19 @@ class Layout extends React.Component {
     this.setState({ isDragging: false });
   }
 
-  onResize(pane, mode) {
-    return size => {
+  onResize(pane, mode, size) {
+    if (pane && mode && size) {
```

Comment:
Looks a bit hacky to me. Let's extract the part that updates state into a separate function.

In fact, the state updates should be throttled (because they provide visual feedback), and the `saveSizes` part should be debounced (because it only makes sense to store sizes when they are final)

Reply:
Sounds good!

Reply:
@Hypnosphi seems that `size` is used to set `defaultSize` and that value is only changed when you save it to local storage. Thus I was getting some weird behavior when I was debouncing. So I opted to throttle both.

Reply:
I also made saveSize on a smaller interval for smoother user experience